File: src/main/java/com/timbuchalka/Location.java
package com.timbuchalka;

import java.util.HashMap;
import java.util.Map;

public class Location {
    private final int locationID;
    private final String description;
    private final HashMap exits;

    public Location(int locationID, String description) {
        this.locationID = locationID;
        this.description = description;
        this.exits = new HashMap();
        this.exits.put("Q", 0);
    }

    public int getLocationID() {
        return locationID;
    }

    public String getDescription() {
        return description;
    }

    public Map<String, Integer> getExits() {
        return new HashMap(exits);
    }

    protected void addExit(String direction, int location) {
        exits.put(direction, location);
    }
}

File: src/silverstar/learningkotlin/textadventure/autoconverted/Location.kt
package silverstar.learningkotlin.textadventure.autoconverted


class Location(val locationID: Int, val description: String) {
    private val exits: MutableMap<String?, Int?>
    fun getExits(): Map<String?, Int?> {
        return HashMap(exits)
    }

    fun addExit(direction: String?, location: Int) {
        exits[direction] = location
    }

    init {
        exits = HashMap()
        exits["Q"] = 0
    }
}